### PR TITLE
Rename ∅ᵐ → ∅

### PR DESCRIPTION
**Pull request by williamdemeo**
# Description

Addresses one item of #282.

The task is to rename `∅ᵐ` as `∅` using type classes.

# Checklist

- [X] Commit sequence broadly makes sense and commits have useful messages
- [X] Code is formatted according to [CONTRIBUTING.md](https://github.com/input-output-hk/formal-ledger-specifications/blob/master/CONTRIBUTING.md)
- [X] Self-reviewed the diff

### CODE REVIEW

File: src/Axiom/Set.agda [hunk 1]
```diff
@@ -167,13 +167,13 @@ record Theory {ℓ} : Type (sucˡ ℓ) where
   ∈-unions : {a : A} {U : Set (Set A)} → (∃[ T ] T ∈ U × a ∈ T) ⇔ a ∈ proj₁ (unions U)
   ∈-unions = proj₂ $ unions _
 
-  ∅ : Set A
-  ∅ = fromList []
+  ﹛﹜ : Set A
```
Comment (williamdemeo) [inline src/Axiom/Set.agda:170]: These are small curly braces obtained from `\(` (or `\)`) and then hitting `C-f` a bunch of times.

**Review by WhatisRT — CHANGES_REQUESTED** [review]
Sorry, the intent here was that both the empty set and empty map would be denoted by `∅`, via a typeclass.

**Comment by williamdemeo** [comment]
@WhatisRT the renaming now uses type classes so that both empty sets and empty maps can be called `∅`. It seems acceptable for now, but I suspect I could have arranged it so that I don't need to import the type classes whenever I use them; e.g., by putting the instances inside the `Set` or `Map` module.  (I did try that initially, but I couldn't get it to work because of circular dependencies.  When more of the items in the issue #282 are done, I'll revisit this.)

File: src/Axiom/Set.agda [hunk 2]
```diff
@@ -167,13 +167,13 @@ record Theory {ℓ} : Type (sucˡ ℓ) where
   ∈-unions : {a : A} {U : Set (Set A)} → (∃[ T ] T ∈ U × a ∈ T) ⇔ a ∈ proj₁ (unions U)
   ∈-unions = proj₂ $ unions _
 
-  ∅ : Set A
-  ∅ = fromList []
+  ∅ˢ : Set A
+  ∅ˢ = fromList []
 
-  ∅-strongly-finite : strongly-finite {A} ∅
+  ∅-strongly-finite : strongly-finite {A} ∅ˢ
```
Comment (williamdemeo) [inline src/Axiom/Set.agda:173]: The goal of this PR is to rename `∅ ` to `∅`.  To that end, I had to change `∅` to `∅ˢ` in this file (and in a few other places) to accommodate the new type classes `HasEmptySet` and `HasEmptyMap`.  Fortunately, I am able to limit the superscripts to blocks of code in `.agda` files or *hidden* blocks in `.lagda` files.

File: src/Interface/HasEmptySet.agda [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+open import Axiom.Set using (Theory)
+
+module Interface.HasEmptySet {ℓ} (th : Theory {ℓ}) where
+
+open Theory th using () renaming (Set to ℙ_)
+
+record HasEmptySet (A : Set ℓ) : Set ℓ where
+  field
+    ∅ : ℙ A
```
Comment (WhatisRT) [inline src/Interface/HasEmptySet.agda:11]:
```suggestion
record HasEmptySet (A : Set ℓ) : Set ℓ where
  field
    ∅ : A

instance
  _ : {A : Set} → HasEmptySet (ℙ A)
  _ = record { ∅ = ∅ˢ }

  _ : {A B : Set} → HasEmptySet (Map A B)
  _ = record { ∅ = ∅ᵐ }
```

**Review by WhatisRT — CHANGES_REQUESTED** [review]
This is much better, but you don't need two type classes. I've made a suggestion below, and then you can just use `∅` if you reexport `Interface.HasEmptySet` from `Ledger.Prelude`

File: src/Interface/HasEmptySet.agda (hunk 3 above)
Comment (williamdemeo) [reply src/Interface/HasEmptySet.agda:11]: Okay, thanks, I'll try it.

Comment (williamdemeo) [reply src/Interface/HasEmptySet.agda]: @WhatisRT This commit creates a lot of cyclic module dependencies in the set theory files, but since those are `.agda` files and won't show up in the pdf, I can choose to not import `HasEmptySet` there and simply add superscripts to the emptyset symbol in those modules.  (HasEmptySet now needs to import both `Axiom/Set` *and* `Axiom/Map`, so these cannot use `HasEmptySet`.)

That's fine, but now the problem is that `Set` has type `Type l -> Type l` whereas `Map` has type `Type -> Type -> Type`, so the levels don't line up in your unified module `HasEmptySet {l}`.  I'll keep working on it and try to resolve this.

Comment (williamdemeo) [reply src/Interface/HasEmptySet.agda]: I think I figured it out.  Will push the fix soon.

File: src/Axiom/Set/Map.agda [hunk 4]
```diff
@@ -58,7 +58,7 @@ record IsLeftUnique (R : Rel A B) : Type where
   field isLeftUnique : left-unique R
 
 instance
-  ∅-left-unique : IsLeftUnique {A = A} {B = B} ∅
+  ∅-left-unique : IsLeftUnique {A = A} {B = B} ∅ˢ
```
Comment (williamdemeo) [inline src/Axiom/Set/Map.agda:61]: It should be okay to leave superscripts on the emptyset symbol in `.agda` files, since they won't appear in the pdf.

Comment (williamdemeo) [reply src/Axiom/Set/Map.agda:61]: Also, in some cases---namely, in the submodules of `Axiom.Set`---we *must* use superscripts; we can't use the type classes there because that causes a cyclic module dependency since the type class imports `Axiom.Set` and `Axiom.Map`.

File: src/Ledger/Deleg.lagda [hunk 5]
```diff
@@ -10,7 +10,7 @@ module Ledger.Deleg (gs : _) (open GovStructure gs) where
 
 open import Ledger.GovernanceActions gs
 \end{code}
-\begin{figure*}[h]
+\begin{figure*}[h!]
```
Comment (williamdemeo) [inline src/Ledger/Deleg.lagda:13]: Fixing figure placement; I don't think this deserves its own PR.

File: src/Ledger/Transaction.lagda [hunk 6]
```diff
@@ -151,17 +151,14 @@ the transaction body are:
...
-\emph{Abstract functions}
-\begin{code}
 \end{code}
 \end{AgdaSuppressSpace}
 \caption{Definitions used in the UTxO transition system}
 \label{fig:defs:utxo-shelley}
 \end{figure*}
 
 \begin{figure*}[h]
+\emph{Abstract functions}
```
Comment (williamdemeo) [inline src/Ledger/Transaction.lagda:161]: Fixing figures; I don't think this deserves a separate PR.

**Review by WhatisRT — APPROVED** [review]
I've added a cleanup commit on top of this
